trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/tracehandling/Mcr.java

```diff
@@ -102,7 +102,8 @@ private McrTraceCheckResult<L> exploreInterleavings(final List<L> initialTrace)
 		while (counterexample != null) {
 			mLogger.info("---- MCR iteration " + iteration + " ----");
 			iteration++;
-			result = mResultProvider.getResult(counterexample);
+			// TODO Is it correct to not pass any control locations here?
+			result = mResultProvider.getResult(counterexample.getWord(), null);
```

Comment:
@schuessf I am not sure, but perhaps you remember? 

Independent of this, passing null just seems wrong.

Reply:
I don't quite remember this, but here `StrategyModuleMcr::getResult` is called, which calls `StrategyFactory::constructStrategy` with the same counterexample, so I think we should not pass null here, but the whole counterexample.

Reply:
The `counterexample` here is a `NestedRun` returned from the emptiness check of the must-happens-before automaton (`mhbAutomaton`). Since it is not a run of the program, I did not know which sequence of control configurations should be used here.

Reply:
As it is now passed as `Counterexample` it should be ok anyways.